packages/react-templates/src/components/Select/MultiTypeaheadSelect.tsx

```diff
@@ -0,0 +1,348 @@
+  placeholder = 'Select an option',
+  noOptionsFoundMessage = (filter) => `No results found for "${filter}"`,
+  isDisabled,
+  toggleWidth,
+  toggleProps,
+  ...props
+}: MultiTypeaheadSelectProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<(string | number)[]>(
+    (initialOptions?.filter((o) => o.selected) ?? []).map((o) => o.value)
+  );
+  const [inputValue, setInputValue] = React.useState<string>();
+  const [selectOptions, setSelectOptions] = React.useState<MultiTypeaheadSelectOption[]>(initialOptions);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItemId, setActiveItemId] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+
+  const NO_RESULTS = 'no results';
+
+  const openMenu = () => {
+    onToggle && onToggle(true);
+    setIsOpen(true);
+  };
+
+  React.useEffect(() => {
+    let newSelectOptions: MultiTypeaheadSelectOption[] = initialOptions;
+
+    // Filter menu items based on the text input value when one exists
+    if (inputValue) {
+      newSelectOptions = initialOptions.filter((option) =>
+        String(option.content).toLowerCase().includes(inputValue.toLowerCase())
+      );
+
+      // When no options are found after filtering, display 'No results found'
+      if (!newSelectOptions.length) {
+        newSelectOptions = [
+          {
+            isAriaDisabled: true,
+            content:
+              typeof noOptionsFoundMessage === 'string' ? noOptionsFoundMessage : noOptionsFoundMessage(inputValue),
+            value: NO_RESULTS
+          }
+        ];
+      }
+
+      // Open the menu when the input value changes and the new value is not empty
+      openMenu();
+    }
+
+    setSelectOptions(newSelectOptions);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inputValue, initialOptions]);
+
+  React.useEffect(
+    () => setSelected((initialOptions?.filter((o) => o.selected) ?? []).map((o) => o.value)),
+    [initialOptions]
+  );
+
+  const setActiveAndFocusedItem = (itemIndex: number) => {
+    setFocusedItemIndex(itemIndex);
+    const focusedItem = selectOptions[itemIndex];
+    setActiveItemId(focusedItem.value as string);
+  };
+
+  const resetActiveAndFocusedItem = () => {
+    setFocusedItemIndex(null);
+    setActiveItemId(null);
+  };
+
+  const closeMenu = () => {
+    onToggle && onToggle(false);
+    setIsOpen(false);
+    resetActiveAndFocusedItem();
+    setInputValue('');
+  };
+
+  const onInputClick = () => {
+    if (!isOpen) {
+      openMenu();
+    } else if (!inputValue) {
+      closeMenu();
+    }
+  };
+
+  const selectOption = (
+    _event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<HTMLInputElement> | undefined,
+    option: string | number
+  ) => {
+    const selections = selected.includes(option) ? selected.filter((o) => option === o) : [...selected, option];
```

Comment:
```suggestion
    const selections = selected.includes(option) ? selected.filter((o) => option !== o) : [...selected, option];
```

Reply:
The true branch here should be selected.filter((o) => option !== o). Currently it's resetting the selections array when an option is deselected.